cli/main.go

```diff
@@ -0,0 +1,344 @@
+
+	request := &proto.JobCreateRequest{
+		CPU:              cpu,
+		MemBytes:         memBytes,
+		IoBytesPerSecond: ioBytesPerSecond,
+		Command:          command,
+		Args:             args,
+	}
+
+	response, err := client.Start(ctx, request)
+	if err != nil {
+		return fmt.Errorf("error starting job: %v", err)
+	}
+
+	fmt.Print(response.Id)
+
+	return nil
+}
+
+func status(client proto.JobWorkerClient, jobId string) error {
+	job := &proto.JobRequest{
+		Id: jobId,
+	}
+
+	response, err := client.Status(context.Background(), job)
+	if err != nil {
+		return fmt.Errorf("failed to get status: %w", err)
+	}
+
+	fmt.Printf("Jod:%s has status: %s. exitCode:%d, exitReason:%s\n",
+		jobId,
+		response.GetStatus(),
+		response.GetExitCode(),
+		response.GetExitReason())
+
+	return nil
+}
+
+func stream(client proto.JobWorkerClient, jobId string) error {
+	// Initiate the stream with a context that supports cancellation.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	request := &proto.JobRequest{
+		Id: jobId,
+	}
+	response, err := client.Stream(ctx, request)
+	if err != nil {
+		return fmt.Errorf("error creating stream: %v", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, os.Interrupt, syscall.SIGTERM)
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		select {
+		case <-ctx.Done():
+			return
+		case s := <-sigCh:
+			log.Printf("got signal %v, attempting graceful shutdown", s)
+			cancel()
+		}
+	}()
+
+	for {
+		output, err := response.Recv()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				fmt.Print(string(output.GetContent()))
+				break
+			}
+			return fmt.Errorf("failed to receive output: %w", err)
+		}
+
+		fmt.Print(string(output.GetContent()))
+	}
+
+	return nil
+}
+
+func stop(client proto.JobWorkerClient, jobId string) error {
+	job := &proto.JobRequest{
+		Id: jobId,
+	}
+
+	response, err := client.Status(context.Background(), job)
+	if err != nil {
+		return fmt.Errorf("failed to get status: %w", err)
+	}
+
+	fmt.Printf("Jod:%s has status: %s, exitCode:%d, exitReason:%s\n",
+		jobId,
+		response.GetStatus(),
+		response.GetExitCode(),
+		response.GetExitReason())
+
+	return nil
+}
+
+func createClient(cCtx *cli.Context) (proto.JobWorkerClient, error) {
+	host := cCtx.String(commandFlagHost)
+	caCert := cCtx.String(commandFlagCertificate)
+	clientCert := cCtx.String(commandFlagClientCertificate)
+	clientKey := cCtx.String(commandFlagClientPrivateKey)
+
+	return getClient(host, caCert, clientCert, clientKey)
+}
+
+func getClient(host, caCertPath, clientCertPath, clientKeyPath string) (proto.JobWorkerClient, error) {
+	tlsCredentials, err := loadTLSCredentials(caCertPath, clientCertPath, clientKeyPath)
+	if err != nil {
+		log.Fatalf("failed to load TLS credentials: %v", err)
+	}
+
+	clientConnection, err := grpc.NewClient(host, grpc.WithTransportCredentials(tlsCredentials))
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect: %w", err)
+	}
```

Comment:
the client should be closed somewhere

Reply:
<img width="926" alt="Screenshot 2024-09-09 at 2 41 48 PM" src="https://github.com/user-attachments/assets/6ac3f275-7e71-48ca-98fa-ca85fd245492">